service/src/main/java/com/commercetools/pspadapter/payone/util/PaymentUtil.java

```diff
@@ -0,0 +1,25 @@
+package com.commercetools.pspadapter.payone.util;
+
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.payments.Transaction;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Optional.ofNullable;
+
+public final class PaymentUtil {
+
+    public static Optional<Transaction> getTransactionById(@Nullable Payment payment, @Nullable String transactionId) {
+        return ofNullable(payment)
+                .map(Payment::getTransactions)
+                .map(List::stream)
+                .flatMap(transactions -> transactions
+                        .filter(tr -> tr.getId().equals(transactionId))
```

Comment:
is it possible here the a payment might have a `null` transaction?

Reply:
theoretically yes, actually this would mean a bug in a platform or JVM SDK

Reply:
I would still add a filter for null transactions. But you know better about the service if it's significant or not.